cmake-modules/AddGoogleTest.cmake

```diff
@@ -15,7 +15,7 @@ if(CMAKE_VERSION VERSION_LESS 3.11)
     download_project(
         PROJ                googletest
 		GIT_REPOSITORY      https://github.com/google/googletest.git
-		GIT_TAG             release-1.11.0
+		GIT_TAG             release-1.12.1
```

Comment:
This will require updating our cgmanifest.json sha as well to point to the correct commit:
https://github.com/Azure/azure-sdk-for-cpp/blob/2c4900efaa065b50c296059b070ef69bae316f55/sdk/core/azure-core/cgmanifest.json#L45-L51

https://github.com/google/googletest/commit/58d77fa8070e8cec2dc1ed015d66b454c8d78850

Reply:
@ahsonkhan  Thank you sir , much appreciated.